CSS: Try custom properties using location-aware naming scheme

ryelle opened the pull request:
This PR tries replacing the colors in core WP CSS with custom properties, based on the location of the color, `--wp-admin--page--background`, `--wp-admin--heading--color`, etc. More [naming discussion is in progress](https://docs.google.com/document/d/1Z0Wo1W0ZQwFJYPcJ8GlsubA9uKhcFdT8ifYnso82z9I/edit) on google docs.

Replacing colors and creating names here is a manual process, deciding what to name a color and how to set it up in the collection of existing names (should it inherit from another value, like the input border inherits from the surface border, for example). The intent is to come up with a minimum set of variables that CSS-authors can use to create custom admin color schemes, including dark schemes and high contrast schemes.

---

I also added a stylelint config to flag any hex color values, which should make it easier to catch all colors. I don't think the RGBA rule is working. This would not be merged into core.

ryelle committed:
Update buttons

ryelle committed:
Update borders

ryelle committed:
Contextual help

ryelle committed:
list table text

ryelle committed:
Nav tabs

ryelle committed:
Star rating icons

ryelle committed:
Plugin install modal

ryelle committed:
Tweak variable names

ryelle committed:
Add alternate row background

ryelle committed:
Move properties to a new stylesheet

ryelle committed:
Update filters

ryelle committed:
Notices

ryelle committed:
Drop in a stylelint to flag color values

ryelle committed:
Admin menu colors

ryelle committed:
WIP: ectoplasm demo

ryelle committed:
Admin bar styles

ryelle committed:
Update hex regex in stylelint config

laras126 commented:
A few questions - 

1. Do we want to provide fallback values? I saw not all custom properties had them in the PR so far, but definitely something we should standardize.
2. Some values are used in wp-includes as well - e.g. #4f94d4 - are these in scope for color schemes?

ryelle commented:
> Do we want to provide fallback values? I saw not all custom properties had them in the PR so far, but definitely something we should standardize.

I wasn't adding them because I think we can be sure the values will be defined, and wouldn't want to fall back to the wrong value if there's a custom color scheme. I don't have a strong opinion on this either way, though.

> Some values are used in wp-includes as well - e.g. #4f94d4 - are these in scope for color schemes?

Yeah, we'll want to update the `wp-includes` CSS too. There's some vendor CSS (like from Gutenberg) that we can leave alone, but pretty much anything in `wp-includes/css/*.css` should get looked at.

ryelle committed:
Merge branch 'master' into try/custom-properties

ryelle committed:
Update the `admin-menu` property names

ryelle committed:
Update scheme colors

ryelle committed:
Misc property updates

ryelle committed:
Add "editor color", tbd

ryelle committed:
Update plugin & theme editor

ryelle committed:
Update the icon color

Commit:
First pass at CSS custom props for wp-login.php

0aveRyan commented:
PR for `login.css` is #4 🕺

Commit:
Try a common set of theme-agnostic properties

use --theme-- over --common--, restore rating val

circlecube committed:
Add custom prop css colors to install page styles

Repeating some colors as the existing matches had names that were specific enough that using them didn't make sense. Also adding some others that were not already existing which will hopefully be able to be consolidated. I included the `colors` file as a dependency for the install css which broke the layout, hence the `body { height: auto; }` addition. Idelaly this page would be rebuilt with some more consistent page structure, but that's beyond scope of this PR.

ryelle committed:
Merge branch 'master' into try/custom-properties

ryelle committed:
Update accordion backgrounds & borders

ryelle committed:
Update sorthelper (deprecated?)

ryelle committed:
Remove from common, same CSS lives in widgets.css

ryelle committed:
Update highlight & bar classes

ryelle committed:
Bring back the 3rd, mid-level border color and create 3 "steps" of border properties

ryelle committed:
Update all `rgba` colors

ryelle committed:
Bring back missing value

ryelle committed:
Merge branch 'master' into try/custom-properties

ryelle committed:
Update remainder of admin-bar.css

0aveRyan committed:
Update src/wp-admin/css/login.css

Missing dash in --wp-admin--page--background

Co-authored-by: Kelly Dwan <ryelle@users.noreply.github.com>

circlecube committed:
site-icon css color update to custom-property

morganisok committed:
use custom properties in the code-editor css file

circlecube committed:
migrate revisions colors to custom properties

morganisok committed:
Update colors in list-tables CSS

Fixes #11

ryelle committed:
Merge branch 'master' into try/custom-properties

ryelle committed:
Merge pull request #8 from circlecube/try/custom-properties-site-icon

Custom properties, site-icon file

Commit:
swap anchor for link in property name

Commit:
double-dash parent loc and property loc

Commit:
Resolve merge conflicts in custom-properties.css

ryelle committed:
Merge branch '0aveRyan-try/custom-properties-common' into try/custom-properties

rgadon107 committed:
Custom properties for color-picker colors

circlecube committed:
Fix formatting issues

circlecube committed:
Fix formatting issues

circlecube committed:
Fix formatting issues

circlecube committed:
Merge branch 'try/custom-properties' into try/custom-properties

circlecube committed:
Fix formatting issues

circlecube committed:
Merge branch 'try/custom-properties' of github.com:circlecube/wordpress-develop into try/custom-properties

circlecube committed:
use existing surface properties in place rather than adding extra vars per use.

circlecube committed:
Merge branch 'try/custom-properties' into try/custom-properties-revisions

ryelle committed:
Merge branch 'wpalchemist-code-editor' into try/custom-properties

Fixes #9

ryelle committed:
Merge branch 'try/custom-properties-revisions' of https://github.com/circlecube/wordpress-develop into pr-10

ryelle committed:
Fix some naming schemes, add defaults for the slider properties

Combines sub-locations into single location tokens; and creates default `slider-handle--background` etc, inheriting from the existing properties.

ryelle committed:
Use the `--wp-admin--surface--box-shadow` property

ryelle committed:
Merge pull request #10 from circlecube/try/custom-properties-revisions

migrate revisions colors to custom properties

ryelle committed:
Merge branch 'master' into try/custom-properties

circlecube committed:
update wp-pointer colors to use custom properties

circlecube committed:
Merge branch 'try/custom-properties' into try/custom-properties

ryelle committed:
Merge branch 'master' into try/custom-properties

ryelle committed:
Merge branch 'list-tables' into try/custom-properties

ryelle committed:
Merge branch 'try/custom-properties' into try/custom-properties-login

ryelle committed:
Merge pull request #4 from 0aveRyan/try/custom-properties-login

CSS Custom Properties: Login

ryelle committed:
Merge branch 'try/custom-properties' into try/custom-properties

ryelle committed:
Merge pull request #6 from circlecube/try/custom-properties

Add custom prop css colors to install page styles

ryelle committed:
Update src/wp-includes/css/wp-pointer.css

ryelle committed:
Merge branch 'try/custom-properties' into try/custom-properties-wp-pointer

ryelle committed:
Merge pull request #17 from circlecube/try/custom-properties-wp-pointer

update wp-pointer colors to use custom properties

ryelle committed:
Merge modified PR #12

circlecube committed:
Try/custom properties nav menus (#13)

* replace nav-menu colors with custom properties

* Use existing custom properties

Co-authored-by: Kelly Dwan <ryelle@users.noreply.github.com>

ryelle committed:
Clean up login & install properties

ryelle committed:
Merge branch 'master' into try/custom-properties

# Conflicts:
#	src/wp-admin/css/list-tables.css
#	src/wp-admin/css/nav-menus.css
#	src/wp-includes/css/wp-pointer.css

circlecube committed:
update widgets colors to use custom properties (#16)

Update custom properties to use mostly existing properties

circlecube committed:
migrate `wp-auth-check.css` colors to custom properties. (#19)

Co-authored-by: Kelly Dwan <ryelle@users.noreply.github.com>

ryelle committed:
Style Loader: Add new CSS dependency for `custom-properties` file, revert colors changes (#22)

* Bring back legacy color CSS handling

* Add a custom-properties style dependency for our color variable CSS file

* Remove dependencies from `colors`, in favor of explicitly calling `colors` after `wp-admin`

0aveRyan committed:
Custom Properties: `edit.css` (#25)

* admin edit custom color properties

* Update src/wp-admin/css/edit.css

* Update src/wp-admin/css/edit.css

Co-authored-by: Dave Ryan <dave@bluehost.com>
Co-authored-by: Kelly Dwan <ryelle@users.noreply.github.com>

ryelle committed:
Fix missing custom property

0aveRyan committed:
Custom Properties: `media.css` (#26)

* wp admin media custom color properties

* media custom properties missed in first pass

Co-authored-by: Dave Ryan <dave@bluehost.com>

ryelle committed:
Merge branch 'master' into try/custom-properties

ryelle committed:
Move admin-menu properties into the main stylesheet

joyously commented on src/wp-admin/css/install.css:
```diff
@@ -1,40 +1,39 @@
 	max-width: 700px;
 	-webkit-font-smoothing: subpixel-antialiased;
-	box-shadow: 0 1px 1px rgba(0, 0, 0, 0.04);
+	box-shadow: 0 1px 1px rgb(var(--wp-admin--shadow) / 0.04);
```
What is `rgb(var(--wp-admin--shadow) / 0.04)`  ??

joyously commented on src/wp-admin/css/login.css:
```diff
@@ -27,14 +27,12 @@ a {
-		0 0 0 1px #4f94d4,
-		0 0 2px 1px rgba(79, 148, 212, 0.8);
+	color: var(--wp-admin--link--color--hover);
+	box-shadow: var(--wp-admin--box-shadow--focus);
```
Are you doing just colors or entire box shadows?

ryelle replied:
Still open to discussion, but in places where there are multiple box shadows, I think it makes sense to declare the whole value. For example, `--wp-admin--box-shadow--focus` is a pretty common box-shadow on focused items, and there is really no reason to control one of the shadow value but not the other, so for simplicity, it's one custom property.

joyously commented on src/wp-admin/css/login.css:
```diff
@@ -44,20 +42,20 @@ p {
-	background-color: #fff;
-	box-shadow: 0 1px 1px 0 rgba(0, 0, 0, 0.1);
+	background-color: var(--wp-admin--surface--background);
+	box-shadow: 0 1px 1px 0 rgb(var(--wp-admin--shadow) / 0.1);
```
Does `rgb(var(--wp-admin--shadow) / 0.1)`  work?

ryelle replied:
Yes, this is the space-separated format which was introduced with CSS Colors level 4 - it ends up resolving to `rgb(0 0 0 / 0.1)`, equivalent to `rgba(0, 0, 0, 0.1);`. I wouldn't pay too much attention to the format here, how to handle these transparent values is also an open discussion.

joyously replied:
Ah, I haven't kept up with that one. I see it's at 92.9% at caniuse.
It looks crazy. I think you have to consider the format, since this would only work with some values. If the properties need to be interchangeable, the format _is_ important.

joyously commented on src/wp-admin/css/login.css:
```diff
@@ -134,9 +132,9 @@ p {
-	box-shadow: 0 1px 3px rgba(0, 0, 0, 0.04);
+	background: var(--wp-admin--surface--background);
+	border: 1px solid var(--wp-admin--surface--border);
+	box-shadow: 0 1px 3px rgb(var(--wp-admin--shadow) / 0.04);
```
Another one? `rgb(var(--wp-admin--shadow) / 0.04)`

ryelle committed:
Remove redundant color

ryelle committed:
Merge branch 'master' into try/custom-properties

# Conflicts:
#	src/wp-admin/css/common.css

0aveRyan committed:
Custom Properties: `site-health.css` (#27)

* site health colors custom properties

* merge try/custom-properties and resolve conflicts

* move entire box-shadow declaration into property

* Remove duplicated properties

* Use the "modern tabbed page" property for the box shadow

Co-authored-by: Dave Ryan <dave@bluehost.com>
Co-authored-by: Kelly Dwan <ryelle@users.noreply.github.com>
